HOSTEDCP-1184: Document IPv6/IPv4/DualStack deployments for Hypershift in Baremetal (#3008)
openshift/hypershift: docs/content/how-to/mce/Dual/hostedcluster/hostedcluster.md

> In this section we will focus in all the related objects which they will be necessary to achieve a Disconnected Hosted Cluster deployment.

Reviewer: ```suggestion
In this section, we will focus in on all the related objects necessary to achieve a Disconnected Hosted Cluster deployment.
```

**Premises**:

- HostedCluster Name: `hosted-dual`
- HostedCluster Namespace: `clusters`
- Disconnected: `true`
- Network Stack: `Dual`

## Openshift Objects

### Namespaces

> The operator in a usual situation will be in charge of create the HCP (HostedControlPlane) namespace, but in this case we want to include all the objects prior the operator to reconcicle over the HostedCluster object. This way when the operator begins the reconcilliation, it will find all the objects in place.

Reviewer: ```suggestion
The hypershift operator is usually in charge of creating the HCP (HostedControlPlane) namespace, but in this case, we want to include all the objects prior to the hypershift operator reconciling over the HostedCluster object. This way when the hypershift operator begins the reconciliation, it will find all the objects in place.
```

```yaml
---
apiVersion: v1
kind: Namespace
metadata:
  creationTimestamp: null
  name: clusters-hosted-dual
spec: {}
status: {}
---
apiVersion: v1
kind: Namespace
metadata:
  creationTimestamp: null
  name: clusters
spec: {}
status: {}
```

> We will not create object by object but, concatenante all of them in the same file and apply them with just 1 command.

Reviewer: ```suggestion
We will not create all of the objects one by one but will concatenate all of them in the same file and apply them with just one command.
```

### ConfigMap and Secrets

These are the ConfigMaps and Secrets which we will include in the HostedCluster deployment.

```yaml
---
apiVersion: v1
data:
  ca-bundle.crt: |
    -----BEGIN CERTIFICATE-----
    -----END CERTIFICATE-----
kind: ConfigMap
metadata:
  name: clusters-hosted-dual-ca
  namespace: clusters
---
apiVersion: v1
data:
  .dockerconfigjson: xxxxxxxxx
kind: Secret
metadata:
  creationTimestamp: null
  name: hosted-dual-pull-secret
  namespace: clusters
---
apiVersion: v1
kind: Secret
metadata:
  name: sshkey-cluster-hosted-dual
  namespace: clusters
stringData:
  id_rsa.pub: ssh-rsa xxxxxxxxx
---
apiVersion: v1
data:
  key: nTPtVBEt03owkrKhIdmSW8jrWRxU57KO/fnZa8oaG0Y=
kind: Secret
metadata:
  creationTimestamp: null
  name: hosted-dual-etcd-encryption-key
  namespace: clusters
type: Opaque
```

### RBAC Roles

> This is not mandatory, but it allow us to have the Assisted Service Agents located in the same HostedControlPlane namespace than the HostedControlPlane and still be managed by CAPI.

Reviewer: ```suggestion
This is not mandatory, but it allows us to have the Assisted Service Agents located in the same HostedControlPlane namespace than the HostedControlPlane and still be managed by CAPI.
```

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  creationTimestamp: null
  name: capi-provider-role
  namespace: clusters-hosted-dual
rules:
- apiGroups:
  - agent-install.openshift.io
  resources:
  - agents
  verbs:
  - '*'
```

### Hosted Cluster

This is the HostedCluster Object

```yaml
apiVersion: hypershift.openshift.io/v1beta1
kind: HostedCluster
metadata:
  name: hosted-dual
  namespace: clusters
spec:
  additionalTrustBundle:
    name: "clusters-hosted-dual-ca"
  olmCatalogPlacement: guest

>   imageContentSources:

Reviewer: Do we need this here? This is for mirror registry sources for workloads on the guest cluster rather than part of the setup of the hosted cluster.

Author: I think that it's worth to mention it. I can specify it in the documentation itself.

ingress-operator-6548dc758b-f9gtg                     1/2     Running   0          94s
konnectivity-agent-7767cdc6f5-tw782                   1/1     Running   0 	       95s
kube-apiserver-7b5799b6c8-9f5bp                       4/4     Running   0 	       3m7s
kube-controller-manager-5465bc4dd6-zpdlk              1/1     Running   0 	       44s
kube-scheduler-5dd5f78b94-bbbck                       1/1     Running   0 	       2m36s
machine-approver-846c69f56-jxvfr                      1/1     Running   0          92s
oauth-openshift-79c7bf44bf-j975g                      2/2     Running   0 	       62s
olm-operator-767f9584c-4lcl2                          2/2     Running   0 	       93s
openshift-apiserver-5d469778c6-pl8tj                  3/3     Running   0 	       2m36s
openshift-controller-manager-6475fdff58-hl4f7         1/1     Running   0 	       95s
openshift-oauth-apiserver-dbbc5cc5f-98574             2/2     Running   0          95s
openshift-route-controller-manager-5f6997b48f-s9vdc   1/1     Running   0          95s
packageserver-67c87d4d4f-kl7qh                        2/2     Running   0          93s
```

> After some time, we will have almost all the pieces in place and the Control Plane operator awaits for the worker nodes to join the cluster, to do so we need to create some more objects, let's talk about the `InfraEnv` and the `BareMetalHost` in the nexts sections.

Reviewer: ```suggestion
After some time, we will have almost all the pieces in place and the Control Plane Operator waits for the worker nodes to join the cluster. To do so, we need to create some more objects. Let's talk about the `InfraEnv` and the `BareMetalHost` in the next sections.
```

> And this is how the HostedCluster looks like:

Reviewer: ```suggestion
And this is how the HostedCluster currently looks:
```

> And this is how the HostedCluster looks like:

Reviewer: I would move L243-248 prior to L241.
